File: src/main/java/com/github/kuro46/embedscript/script/processor/AbstractParser.java
package com.github.kuro46.embedscript.script.processor;

import com.github.kuro46.embedscript.script.ParseException;
import com.google.common.collect.ImmutableList;

public abstract class AbstractParser implements Processor.Parser {
    @Override
    public void prepareBuild(ScriptProcessor processor, MutableScript buffer, String key, ImmutableList<String> matchedValues) throws ParseException {

    }

    @Override
    public void build(ScriptBuilder builder, String key, ImmutableList<String> matchedValues) throws ParseException {
        builder.getScript().putAll(key, matchedValues);
    }
}

File: src/main/kotlin/com/github/kuro46/embedscript/script/processor/AbstractParser.kt
package com.github.kuro46.embedscript.script.processor

abstract class AbstractParser : Processor.Parser {
    override fun prepareBuild(processor: ScriptProcessor, script: MutableScript, key: String, matchedValues: List<String>) {

    }

    override fun build(builder: ScriptBuilder, key: String, matchedValues: List<String>) {
        builder.script.putAll(key, matchedValues)
    }
}
